src/argprocessing.cpp

```diff
@@ -530,6 +531,12 @@ process_option_arg(const Context& ctx,
     return Statistic::none;
   }
 
+  // --analyze changes the default extension too
+  if (arg == "--analyze") {
+    state.found_analyze_opt = true;
```

Comment:
How about adding

```c++
    state.common_args.push_back(args[i]);
```

here, similar to how it's done for `-S`? Then there would be no need to call `compiler_args.push_back("--analyze");` in `process_arg`.

Reply:
fixed